data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,675 @@
+
+In practice, what is considered an effect is an engineering choice. In most circumstances, I/O operations are considered as effects, unless they are ignored. Electromagnetic radiation emitted by the processor when computing a function isn't usually considered a relevant side-effect, except in some security-sensitive contexts. In the OCaml community, as well as in the wider functional programming community, functions are often said to be either [pure](https://en.wikipedia.org/wiki/Pure_function) or impure. The former does not have side effects, the latter does. This distinction makes sense and is useful. Knowing what the effects are, and when are they taking place, is a key design consideration. However, it is important to remember this distinction always assumes some sort of context. Any computation has effects, and what is considered a relevant effect is a design choice.
+
+Since, by definition, effects lie outside function types, a function type can't reflect the effects a function may have. However, it is important to document the intended side effects a function may have. Consider the `Unix.time` function. It returns the number of seconds elapsed since January 1, 1970.
+```ocaml
+# Unix.time ;;
+- : unit -> float = <fun>
+```
+If you're getting an `Unbound module error` in macOS, run this first: `#require "unix";;`.
+
+To produce its result, no data needs to be passed to that function. The result is entirely determined by external factors. If it was passed information, it would not be used. But something must be passed as a parameter to trigger the request of the current time from the operating system.
+
+Since the function must receive data to trigger the computation but the data is going to be ignored, it makes sense to provide the `unit` value `()`. What is discarded is meaningless in the first place.
+
+A similar reasoning applies to functions producing an effect instead of being externally determined or influenced. Consider `print_endline`. It prints the string it was passed to standard output, followed by a line termination.
+```ocaml
+# print_endline;;
+- : string -> unit = <fun>
+```
+Since the purpose of the function is only to produce an effect, it has no meaningful data to return; therefore, again, it makes sense to return the `unit` value.
+
+This illustrates the relationship between functions intended to have side effects and the `unit` type. The presence of the `unit` type does not indicate the presence of side effects. The absence of the `unit` type does not indicate the absence of side effects. But when no data needs to be passed as input or can be returned as output, the `unit` type should be used to indicate it and suggest the presence of side effects.
```

Comment:
```suggestion
This illustrates the relationship between functions intended to have side effects and the `unit` type. The presence of the `unit` type does not indicate the presence of side effects. The absence of the `unit` type does not indicate the absence of side effects. But when no data needs to be applied as input or can be returned as output, the `unit` type should be used to indicate it and suggest the presence of side effects.
```
Again, "passed" or "applied"?

Reply:
I remember we discussed that a couple of weeks ago, and my understanding was that it's called "applying" (officially) in functional languages, even though people understand when called "passing," since that's what it's called in Imperative languages. 

Totally fine to use both! Just checking in case it was necessary/beneficial to reinforce the "official" term. Agree about using both, really.